lib/src/main/java/uk/gov/di/ipv/cri/passport/library/service/KmsRsaDecrypter.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.di.ipv.cri.passport.library.service;
+
+import com.amazonaws.services.kms.AWSKMS;
+import com.amazonaws.services.kms.AWSKMSClientBuilder;
+import com.amazonaws.services.kms.model.DecryptRequest;
+import com.amazonaws.services.kms.model.DecryptResult;
+import com.nimbusds.jose.EncryptionMethod;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWEAlgorithm;
+import com.nimbusds.jose.JWEDecrypter;
+import com.nimbusds.jose.JWEHeader;
+import com.nimbusds.jose.crypto.impl.AlgorithmSupportMessage;
+import com.nimbusds.jose.crypto.impl.ContentCryptoProvider;
+import com.nimbusds.jose.jca.JWEJCAContext;
+import com.nimbusds.jose.util.Base64URL;
+import uk.gov.di.ipv.cri.passport.library.annotations.ExcludeFromGeneratedCoverageReport;
+
+import javax.crypto.spec.SecretKeySpec;
+
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.amazonaws.services.kms.model.EncryptionAlgorithmSpec.RSAES_OAEP_SHA_256;
+import static com.nimbusds.jose.JWEAlgorithm.RSA_OAEP_256;
+
+@ExcludeFromGeneratedCoverageReport
```

Comment:
Ooof. I take it this is excluded because sonar isn't running on the integration tests?

Reply:
Yeah, do we need to exclude the entire class and not just the test constructor?

Reply:
Yeah I don't think sonar includes code coverage from the integration tests. Not sure if it only needs to be on the constructor, I would guess that would cause sonar to complain about coverage of everything else I think?